lib/cylc/task_events_mgr.py

```diff
@@ -663,10 +663,15 @@ def _process_message_succeeded(self, itask, event_time):
                 itask.summary['finished_time'] -
                 itask.summary['started_time'])
         if not itask.state.outputs.all_completed():
-            message = "Succeeded with unreported outputs:"
+            msg = ""
             for output in itask.state.outputs.get_not_completed():
-                message += "\n  " + output
-            LOG.info(message, itask=itask)
+                if output not in [TASK_OUTPUT_EXPIRED,
+                                  TASK_OUTPUT_SUBMIT_FAILED,
+                                  TASK_OUTPUT_FAILED]:
+                    msg += "\n  " + output
```

Comment:
Note on this branch tasks retain these "alternate standard outputs" permanently (but normally in the non-completed state).  They were being added and removed on the fly, which was messy, just to avoid this log message on normal successful task completion.

Reply:
This probably adds a tiny amount to the memory footprint, but it is the right thing to do IMO.